frm/four_roll_mill.cpp

```diff
@@ -171,6 +184,46 @@ main(int argc, char* argv[])
                 visit_data_writer->writePlotData(patch_hierarchy, iteration_num, loop_time);
                 next_viz_dump_time += viz_dump_time_interval;
             }
+
+            // At specified intervals, store hierarchy data for post processing.
+            const bool last_step = !ins_integrator->stepsRemaining();
+            if (dump_postproc_data && last_step)   
+            {
+                output_data(patch_hierarchy, ins_integrator, adv_diff_integrator, iteration_num, loop_time, postproc_data_dump_dirname);
+            }
         }
     } // cleanup dynamically allocated objects prior to shutdown
 } // main
+
+void
+output_data(Pointer<PatchHierarchy<NDIM> > patch_hierarchy,
+                 Pointer<INSVCTwoFluidStaggeredHierarchyIntegrator> ins_integrator,
+                 Pointer<AdvDiffSemiImplicitHierarchyIntegrator> adv_diff_integrator,
+                 const int iteration_num,
+                 const double loop_time,
+                 const string& data_dump_dirname)
+{
```

Comment:
Added this function that creates a hierarchy data dump file.

Reply:
:+1: